src/rpc/mining.cpp

```diff
@@ -544,6 +544,7 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
         pindexPrev = pindexPrevNew;
     }
     CBlock* pblock = &pblocktemplate->block; // pointer for convenience
+    assert(pblock);
```

Comment:
I've now re-read the code and now my reading is that assuming the `if`-branch is taken on the first call then there is no possibility of a `NULL` pointer dereference.

The question then boils down to if `pindexPrev != chainActive.Tip() || (mempool.GetTransactionsUpdated() != nTransactionsUpdatedLast && GetTime() - nStart > 5)` is guaranteed to be true on the first call :-)